utilities/general.py

```diff
@@ -171,3 +180,144 @@ def create_isvc_label_selector_str(isvc: InferenceService, resource_type: str, r
+                 where operator_name is determined by the distribution (rhods-operator for OpenShift AI,
+                 opendatahub-operator for Open Data Hub)
+
+    Returns:
+        List of related images from the CSV
+    """
+    from utilities.infra import (
+        get_product_version,
+        get_operator_distribution,
+    )  # Import here to avoid circular dependency
+
+    if csv_name is None:
+        distribution = get_operator_distribution(client=admin_client)
+        operator_name = "opendatahub-operator" if distribution == "Open Data Hub" else "rhods-operator"
+        csv_name = f"{operator_name}.{get_product_version(admin_client=admin_client)}"
+
+    csvs = list(
+        ClusterServiceVersion.get(
+            dyn_client=admin_client,
+            name=csv_name,
+            namespace=py_config["applications_namespace"],
+        )
+    )
+
+    if not csvs:
+        raise ValueError(f"CSV {csv_name} not found in namespace {py_config['applications_namespace']}")
+
+    csv = csvs[0]  # Get the first (and should be only) CSV instance
+    if len(csvs) > 1:
+        LOGGER.warning(f"Multiple CSV instances found for {csv_name}, using the first one")
+    if not csv.instance:
+        raise ValueError(f"CSV {csv_name} has no instance data")
+
+    return csv.instance.spec.relatedImages
+
+
+def get_pod_images(pod: Pod) -> List[str]:
+    """Get all container images from a pod.
+
+    Args:
+        pod: The pod to get images from
+
+    Returns:
+        List of container image strings
+    """
+    return [container.image for container in pod.instance.spec.containers]
+
+
+def validate_image_format(image: str) -> Tuple[bool, str]:
+    """Validate image format according to requirements.
+
+    Args:
+        image: The image string to validate
+
+    Returns:
+        Tuple of (is_valid, error_message)
+    """
+    if not image.startswith(Resource.ApiGroup.IMAGE_REGISTRY):
+        return False, f"Image {image} is not from {Resource.ApiGroup.IMAGE_REGISTRY}"
+
+    if not re.search(SHA256_DIGEST_PATTERN, image):
+        return False, f"Image {image} does not use sha256 digest"
+
+    return True, ""
+
+
+def get_pods_by_labels(
+    admin_client: DynamicClient,
+    namespace: str,
+    label_selector: str,
+) -> list[Pod]:
+    """
+    Get pods by label selector in a namespace.
+
+    Args:
+        admin_client: The admin client to use for pod retrieval
+        namespace: The namespace to search in
+        label_selector: The label selector to filter pods
+
+    Returns:
+        List of matching pods
+
+    Raises:
+        pytest.fail: If no pods are found
+    """
+    pods = list(
+        Pod.get(
+            dyn_client=admin_client,
+            namespace=namespace,
+            label_selector=label_selector,
+        )
+    )
+    if not pods:
+        pytest.fail(f"No pods found with label selector {label_selector} in namespace {namespace}")
```

Comment:
Since it is an util, we should raise ResourceNotFoundError. Caller can fail then when such exceptions are raised.